### ISSUE TITLE
store-sales-prediction

### ISSUE THREAD

**Issue by ken-wakura-pro** [wakura-mbuya/Pharmaceutical-Sales-prediction#7]
The main objective of this task is to predict daily sales in various stores up to 6 weeks ahead of time. The following steps outline the various sub tasks needed to effectively do this: 

- **Preprocessing**: Preprocess the data into formats that can be fed into a machine learning model
- ** Building models with sklearn pipelines**: Since the problem at hand is a regression problem, I will use tree-based algorithms to build the prediction models. I will also use sklearn pipelines to ensure the modeling is modular and easily reproducible
- **Choose a loss function**: I will need to select a loss function to help me indicate how well the models are performing. I will use need to choose a loss function that is easily interpretable
- **Post Prediction analysis**: Explore the feature importance from our modeling. Creatively deduce a way to estimate the confidence interval of your predictions. 
- **Serialize models**: Serialize the models I have built in order to be able to serve them on a dashboard. I will have the model with a timestamp(eg. 10-08-2020-16-32-31-00.pkl). This is necessary so that I can track predictions from various models.
- **Building model with deep learning**: create a deep learning model of the [Long Short Term Memory](https://arxiv.org/abs/1909.09586) which is a type of [Recurrent Neural Network ](https://arxiv.org/abs/1808.03314). I will use either TensorFlow or Pytorch libraries for modeling. 
- **Using MLFlow to serve the prediction**:  I will use the code snippet provided by mlflow to make inference on the test data
